includes/free-trial/partners/class-wc-calypso-bridge-partner-paypal.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * WC Calypso Bridge Partner PayPal
+ *
+ * @since   x.x.x
+ * @version x.x.x
+ *
+ * This file includes customizations for the sites that were created through /start/paypal on woo.com.
+ * woocommerce_onboarding_profile.partner must get 'paypal'
+ */
+class WC_Calypso_Bridge_Partner_PayPal {
+
+    /**
+     * Class instance.
+     *
+     * @var WC_Calypso_Bridge_Partner_PayPal instance
+     */
+    protected static $instance = false;
+
+    /**
+     * Get class instance
+     */
+    public static function get_instance() {
+        if ( ! self::$instance ) {
+            self::$instance = new self();
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * Constructor.
+     */
+    public function __construct() {
+        // Only for free trials.
+        if ( ! wc_calypso_bridge_is_ecommerce_trial_plan() ) {
```

Comment:
I think we should avoid using the `wc_calypso_bridge_is_ecommerce_trial_plan` condition. In the future the plan provisioned might not be a trial. Also, the behaviour should only be dictated by the `$onboarding_profile['partner']`.

Reply:
That's a key point to consider. We included that condition to avoid diverging from the original implementation, but I soon realized that it would need an update if the creating flow changes at any point, which is indeed very likely. And I totally agree with you that it should only depend on the `onboarding_profile` option.

What do you think about considering this when updating the process to be more dynamic, context here: p1721665079790669-slack-C02JPCHEKDY? Or do you think that we should instead remove this condition before shipping this iteration?

Reply:
We can do it in a clean up PR.

Reply:
Task created to avoid forgetting this: 
- https://github.com/Automattic/dotcom-forge/issues/8413